mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingBatchCoordinator.java

```diff
@@ -142,29 +140,36 @@ private void doBatchWork() throws InterruptedException {
 	}
 
 	private <E> PojoMassIndexingBatchIndexingWorkspace<E, ?> createBatchIndexingWorkspace(
-			PojoMassIndexingIndexedTypeGroup<E> typeGroup) {
+			PojoMassIndexingIndexedTypeGroup<E> typeGroup, BatchCoordinatorContext context) {
 		return new PojoMassIndexingBatchIndexingWorkspace<>(
 				mappingContext, getNotifier(), typeGroup,
 				typeGroup.loadingStrategy(),
-				documentBuilderThreads );
+				documentBuilderThreads,
+				context.sessionContext().tenantIdentifier()
+		);
 	}
 
 	/**
 	 * Operations to do after all subthreads finished their work on index
 	 */
 	private void afterBatch() throws InterruptedException {
 		if ( mergeSegmentsOnFinish ) {
-			Futures.unwrappedExceptionGet( scopeWorkspace.mergeSegments( OperationSubmitter.BLOCKING ) );
+			Futures.unwrappedExceptionGet( coordinatorContexts.iterator().next()
+					.scopeWorkspace().mergeSegments( OperationSubmitter.BLOCKING )
+			);
 		}
 		flushAndRefresh();
-		Futures.unwrappedExceptionGet( agent.preStop() );
-		agent.stop();
-		agent = null;
+		for ( BatchCoordinatorContext context : coordinatorContexts ) {
+			Futures.unwrappedExceptionGet( context.agent().preStop() );
```

Comment:
It would be better to do this in parallel. See `org.hibernate.search.mapper.pojo.work.impl.PojoScopeWorkspaceImpl#doOperationOnTypes` for how to do that properly (basically create an array of futures then use `Futures.unwrappedExceptionGet( CompletableFuture.allOf( futures ) )`).